### PR TITLE
[Word] (Samples) Updating JS patterns in Word articles

### PR DESCRIPTION
**Pull request by AlexJerabek**
This PR updates the JavaScript samples in the Word-specific articles to match the modern JS patterns that use `async`/`await`.

Reviewers, please note that I was unable to successfully run the Word tutorial with Visual Studio under any circumstance. Please let me know if that problem is happening for you too.

### CODE REVIEW

File: docs/word/search-option-guidance.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Use search options to find text in your Word add-in 
 description: 'Learn to use search options in your Word add-in'
```
Comment (ElizabethSamuel-MSFT) [inline docs/word/search-option-guidance.md:3]:
```suggestion
description: Learn to use search options in your Word add-in.
```

File: docs/tutorials/word-tutorial.md [hunk 2]
```diff
@@ -738,15 +725,15 @@ Complete the following steps to define the image that you'll insert into the doc
...
 
             // TODO1: Queue commands to get a reference to the paragraph
             //        that will proceed the table.
 
             // TODO2: Queue commands to create a table and populate it with data.
 
-            return context.sync();
+            await context.sync();
         })
         .catch(function (error) {
```
Comment (ElizabethSamuel-MSFT) [inline docs/tutorials/word-tutorial.md:738]: Should we keep these catch statements?

File: docs/word/word-add-ins-programming-overview.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Word add-ins overview
 description: 'Learn the basics of Word Add-ins.'
```
Comment (ElizabethSamuel-MSFT) [inline docs/word/word-add-ins-programming-overview.md:3]:
```suggestion
description: Learn the basics of Word Add-ins.
```

File: docs/word/word-add-ins-programming-overview.md [hunk 4]
```diff
@@ -24,13 +24,13 @@ The following figure shows an example of a Word add-in that runs in a task pane.
 The Word add-in (1) can send requests to the Word document (2) and can use JavaScript to access the paragraph object and update, delete, or move the paragraph. For example, the following code shows how to append a new sentence to that paragraph.
```
Comment (ElizabethSamuel-MSFT) [inline docs/word/word-add-ins-programming-overview.md:24]:
```suggestion
The Word add-in can (1) send requests to the Word document and (2) use JavaScript to access the paragraph object and update, delete, or move the paragraph. For example, the following code shows how to append a new sentence to that paragraph.
```

**Review by ElizabethSamuel-MSFT — COMMENTED** [review]
What I've found so far. I'll try the quick start and let you know the outcome.

File: docs/tutorials/word-tutorial.md (hunk 2 above)
Comment (AlexJerabek) [reply docs/tutorials/word-tutorial.md:738]: There's no wrapper function like in the Script Lab snippets. We could rewrite the tutorial to have that, but I figured this practice here is good. It shows the action and error handling, which is needed in a complete app. In the other sample code blocks (outside of the tutorial), we're wanting to highlight the API in question, so the catch block only serves as a distraction.

**Comment by aafvstam** [comment]
I reviewed the changes, it contained 3 errors and a small issue in the Start instructions. Created a PR for that on AlexJ-Word branch: #3272 
Expected it to land in this PR but created a separate PR apparently.

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**